python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1479,94 +1343,84 @@ def _execute_asset_backfill_iteration_inner(
+        backfill_id, asset_backfill_data, asset_graph, instance_queryer
+    )
 
-        updated_materialized_subset = get_asset_backfill_iteration_materialized_subset(
-            backfill_id, asset_backfill_data, asset_graph, instance_queryer
-        )
+    materialized_since_last_tick = (
+        updated_materialized_subset - asset_backfill_data.materialized_subset
+    )
+    logger.info(
+        f"Assets materialized since last tick:\n{_asset_graph_subset_to_str(materialized_since_last_tick, asset_graph)}"
+        if not materialized_since_last_tick.is_empty
+        else "No relevant assets materialized since last tick."
+    )
 
-        materialized_since_last_tick = (
-            updated_materialized_subset - asset_backfill_data.materialized_subset
-        )
-        logger.info(
-            f"Assets materialized since last tick:\n{_asset_graph_subset_to_str(materialized_since_last_tick, asset_graph)}"
-            if not materialized_since_last_tick.is_empty
-            else "No relevant assets materialized since last tick."
-        )
+    failed_asset_graph_subset = _get_failed_asset_graph_subset(
+        asset_graph_view,
+        backfill_id,
+        updated_materialized_subset,
+    )
 
-        failed_asset_graph_subset = _get_failed_asset_graph_subset(
-            asset_graph_view,
-            backfill_id,
-            updated_materialized_subset,
-        )
-        candidate_asset_graph_subset = _get_candidate_asset_graph_subset(
-            asset_backfill_data,
-            asset_graph_view,
-            updated_materialized_subset,
-            failed_asset_graph_subset,
-        )
+    failed_and_downstream_subset = _get_failed_and_downstream_asset_graph_subset(
+        backfill_id,
+        asset_backfill_data,
+        asset_graph_view,
+        updated_materialized_subset,
+        failed_asset_graph_subset,
+    )
 
-        failed_and_downstream_subset = _get_failed_and_downstream_asset_graph_subset(
-            backfill_id,
-            asset_backfill_data,
-            asset_graph_view,
-            updated_materialized_subset,
-            failed_asset_graph_subset,
-        )
+    candidate_asset_graph_subset = (
+        asset_backfill_data.target_subset
+        - asset_backfill_data.requested_subset
+        - failed_and_downstream_subset
+    )
 
     logger.info(
         f"Considering the following candidate subset:\n{_asset_graph_subset_to_str(candidate_asset_graph_subset, asset_graph)}"
         if not candidate_asset_graph_subset.is_empty
         else "Candidate subset is empty."
     )
 
-    asset_subset_to_request, not_requested_and_reasons = bfs_filter_asset_graph_view(
-        asset_graph_view,
-        lambda candidate_asset_graph_subset,
-        visited: _should_backfill_atomic_asset_graph_subset_unit(
+    asset_graph_subset_to_request = AssetGraphSubset()
+    not_requested_and_reasons = []
+    for asset_key in asset_graph.toposorted_asset_keys:
```

Comment:
wondering if there are any dragons with non-subsettable assets here - see e.g. https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/asset_graph_view/bfs.py#L175-L177

where we set the level to max of all toposort levels